lib/logstash/filters/geoip.rb

```diff
@@ -84,6 +84,16 @@ class LogStash::Filters::GeoIP < LogStash::Filters::Base
   # number of cache misses and waste memory.
   config :lru_cache_size, :validate => :number, :default => 1000
 
+  # If the ip field is an array: check which one is not a private ip. The first non private ip will be used.
+  config :filter_private_ips, :validate => :boolean, :required => false, :default => true
+
+  # String by which the ip field should be tokenized
+  config :ip_split_symbol, :validate => :string, :required => false, :default => ","
+
+  # list of ip patterns that private ips start with. 
```

Comment:
I'd love to have CIDR notation. If performance becomes a problem, you could do an optimization on common cases (/8, /16, /24) to do substring comparison for those. Could even do that for things like /12 -- My hope was for users to see CIDR notation. In the code itself we can do whatever we like :)